backend/core/src/sms/controllers/sms.controller.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  Body,
+  Controller,
+  HttpException,
+  HttpStatus,
+  Post,
+  Request,
+  UseGuards,
+  UsePipes,
+  ValidationPipe,
+} from "@nestjs/common"
+import { ApiBearerAuth, ApiOperation, ApiTags } from "@nestjs/swagger"
+import { ResourceType } from "../../auth/decorators/resource-type.decorator"
+import { User } from "../../auth/entities/user.entity"
+import { AuthzGuard } from "../../auth/guards/authz.guard"
+import { OptionalAuthGuard } from "../../auth/guards/optional-auth.guard"
+import { AuthContext } from "../../auth/types/auth-context"
+import { defaultValidationPipeOptions } from "../../shared/default-validation-pipe-options"
+import { StatusDto } from "../../shared/dto/status.dto"
+import { SmsDto } from "../dto/sms.dto"
+import { SmsService } from "../services/sms.service"
+
+@Controller("sms")
+@ApiBearerAuth()
+@ApiTags("sms")
+@ResourceType("user")
+@UsePipes(new ValidationPipe(defaultValidationPipeOptions))
+export class SmsController {
+  constructor(private readonly smsService: SmsService) {}
+
+  @Post()
+  @UseGuards(OptionalAuthGuard, AuthzGuard)
+  @ApiOperation({ summary: "Send an SMS", operationId: "send-sms" })
+  async send(@Request() req, @Body() dto: SmsDto): Promise<StatusDto> {
+    // Only admins are allowed to send SMS messages.
+    if (!new AuthContext(req.user as User).user.roles?.isAdmin) {
```

Comment:
I wanted to minimize the changes outside of the additional files I'm adding in this feature, since we'll need to integrate something like this from upstream. The controller addition is really only so that I could test through the Swagger docs, and won't really be needed since we probably won't ever want frontend sending SMS messages.

Maybe it would be better to write a script or something that can exercise the SmsService? I was less sure how to do that though.

Reply:
That makes sense, thanks for explaining! And in that case, I think it sounds good to keep the auth stuff in this file.